tools/pythonpkg/src/python_udf.cpp

```diff
@@ -154,35 +187,74 @@ static scalar_function_t CreateVectorizedFunction(PyObject *function, PythonExce
 			}
 		}
 		// Convert the pyarrow result back to a DuckDB datachunk
-		ConvertPyArrowToDataChunk(python_object, result, state.GetContext(), count);
+		if (count != input_size) {
+			// We filtered out some NULLs, now we need to reconstruct the final result by adding the nulls back
+			Vector temp(result.GetType(), count);
+			// Convert the table into a temporary Vector
+			ConvertPyArrowToDataChunk(python_object, temp, state.GetContext(), count);
+			SelectionVector inverted(input_size);
```

Comment:
`ConvertPyArrowToDataChunk` should also not return any NULL values here - perhaps we can add a test & add verification that no NULL values are returned? Then we can throw an error message instructing the user to change the `NULL_HANDLING` in that case.

Reply:
I believe that's done upstream by `VerifyNullHandling`, but that is DEBUG code, I assume you want this done on release builds?

Reply:
Yes I think doing this on release builds makes sense since this is user input, not an internal function created by us.